bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/ThreadFactoryBuilder.java

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.common;
+
+import static java.util.concurrent.Executors.defaultThreadFactory;
+
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * A builder for {@link ThreadFactory} instances. This builder is intended to be use for creating thread factories to be
+ * used, e.g., when creating {@link Executor}s via the {@link Executors} utility methods.
+ * <p/>
+ * The built {@link ThreadFactory} uses a wrapped {@link ThreadFactory} to create threads (defaulting to
+ * {@link Executors#defaultThreadFactory()}, and then overwrites thread properties as indicated in the build process.
+ *
+ * @author Henning Sudbrock - initial contribution
+ */
+@NonNullByDefault
+public class ThreadFactoryBuilder {
+
+    @Nullable
+    private ThreadFactory wrappedThreadFactory = null;
```

Comment:
Any reason to explicit assign the default value to the references (`null`, `false`)?
IMHO it does not make the code more readable to add stuff that is not necessary.

Reply:
I, personally, like having the default values explicitly set - but do not have a very strong opinion on that. I will change it in the PR.

Reply:
Thank you, we don't "normally" add stuff that is already implicit set (same for default values of annotation attributes, ...)